Lib/test/test_dataclasses/__init__.py

```diff
@@ -2222,7 +2224,10 @@ class C:
             """Lorem ipsum"""
             x: int
 
-        self.assertEqual(C.__doc__, "Lorem ipsum")
+        if support.HAVE_DOCSTRINGS:
+            self.assertEqual(C.__doc__, "Lorem ipsum")
+        else:
+            self.assertEqual(C.__doc__, "C(x: int)")
```

Comment:
`test_docstring_one_field` below already tests that ""C(x:int)" replaces None.  If the replacement code is altered, that and the tests that follows are enough failures.  This test is about not replacing a non-None docstring and should just test that. So perhaps better just put the non-None docstring back when it is stripped by the command line option;
```suggestion
        if C.__doc__ == None:  # If stripped by a "-00" command line option.
            C.__doc__ = "Lorem ipsum"
        self.assertEqual(C.__doc__, "Lorem ipsum")
```
or even better, just condition the test on some docstring being present.
```suggestion
        if C.__doc__ != None:  # If not stripped by a "-00" command line option.
            self.assertEqual(C.__doc__, "Lorem ipsum")
```